### PR TITLE
Break three oversized test files into small, focused ones

### PR DESCRIPTION
**Pull request by stefan-burke**
Three test files had grown well past the ~400-line size we aim for. Each one is now a small folder of files, with one clear subject per file.

Nothing about how the site works has changed. Every check that was there before is still there, word for word — the tests were only moved into new homes, and shared setup was lifted into one helper per folder so it is not written out twice.

**What moved where**

- `test/ui/templates/admin/questions.test.ts` (862 lines) → `questions/` with `list`, `detail`, `answer`, `delete`, `listing-panel`, and `text` files, plus a shared `fixtures.ts`.
- `test/features/admin/auth.test.ts` (651 lines) → `auth/` with `login`, `logout`, and `sessions` files.
- `test/ui/templates/checkin.test.ts` (523 lines) → `checkin/` with `rendering`, `flows`, and `routes` files, plus a shared `helpers.ts`.

**Why it is worth doing**

Smaller files are quicker to read, and they let the mutation checker run a source file against just the handful of tests that cover it instead of one big suite. Both jobs were already written up in `TODO.md`; those two entries are removed here now they are done.

Test counts are unchanged: 69 for the questions pages, 42 for admin sign-in, and 27 for check-in. `deno task precommit` passes.

---
_Generated by [Claude Code](https://claude.ai/code/session_01EdDZCWRXwjqsidfRJFarXC)_

## Summary by CodeRabbit

* **Tests**
  * Expanded coverage for admin login, logout, session management, and session expiration scenarios.
  * Added validation for admin question and answer pages, deletion flows, listing assignments, read-only views, and question text formatting.
  * Added coverage for check-in page rendering, check-in/check-out actions, route validation, authentication, CSRF protection, and edge cases.
  * Reorganized large test suites into focused areas for easier maintenance.

### CODE REVIEW

File: test/ui/templates/admin/questions/fixtures.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+/** The "T-shirt size?" question with Small/Large answers — the canonical radio
+ *  question reused by the question, answer-edit, and answer-delete page tests. */
+export const tShirtQuestion = testQuestion({
+  answers: smallLargeAnswers,
+  id: 1,
+  text: "T-shirt size?",
+});
+
+export const setupQuestionPageTest = setupFeaturePageTest("questions");
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/ui/templates/admin/questions/fixtures.ts]: Fixed in f6957b4 — `setupQuestionPageTest` is now annotated `() => Promise<void>` at the export, matching `setupFeaturePageTest`'s declared return type. Lint, `deno check` on the folder, and the 69 question-page tests all pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/features/admin/auth/login.test.ts [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+      // 6th attempt should be rate limited
+      const response = await handleRequest(await makeRequest());
+      expect(response.status).toBe(302);
+      expectFlash(
+        response,
+        expect.stringContaining("Too many login attempts"),
+        false,
+      );
+    });
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/features/admin/auth/login.test.ts]: Fixed in 99edbd1 — renamed to "redirects with a too-many-attempts flash when rate limited", which is what the 302 + flash assertions actually check. (Pre-existing name that came across in the move, but worth correcting while the file is open.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/features/admin/auth/sessions.test.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+      );
+      const sessionToken = sessionMatch?.[1];
+
+      await handleRequest(
+        mockFormRequest("/admin/sessions", { csrf_token: csrfToken }, cookie),
+      );
+
+      // Verify current session still exists
+      const currentSession = await getSession(sessionToken || "");
+      expect(currentSession).not.toBeNull();
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/features/admin/auth/sessions.test.ts:148]: Fixed in 99edbd1 — `sessionMatch[1]!` with the invariant noted in a comment, and `getSession(sessionToken)` with no `|| ""`. Agreed on the diagnosis point: the fallback turned "no cookie was issued" into a failure that read as "the current session was deleted".

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/ui/templates/admin/questions/answer.test.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+      answer,
+      OWNER_SESSION,
+      "Invalid modifier",
+      aligned,
+      modifiers,
+      null,
+    );
+    expect(html).toContain("Invalid modifier");
+  });
+});
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/ui/templates/admin/questions/answer.test.ts:137]: Fixed in 99edbd1 for both files. `answer.test.ts` now renders the default answer-edit page once in `beforeAll` (after `setupQuestionPageTest`, since the render depends on the feature setting it applies) and the seven matching tests assert against it; the four tests with different arguments — deactivated answer, drifted totals, linked modifier, error message — keep their own renders. `delete.test.ts` does the same per describe block.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/ui/templates/admin/questions/answer.test.ts [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    expect(
+      adminAnswerRecalculatePage(
+        question,
+        answer,
+        snapshot,
+        OWNER_SESSION,
+        undefined,
+        "Selection total recalculated",
+      ),
+    ).toContain("Selection total recalculated");
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/ui/templates/admin/questions/answer.test.ts]: Fixed in 99edbd1 — now "renders an error flash" and "renders a success flash", one `adminAnswerRecalculatePage` call and one assertion each.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/ui/templates/admin/questions/list.test.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    );
+    expect(html).toContain(
+      '<a class="guide-link" href="/admin/guide#questions">',
+    );
+    // The guide link carries its "Questions guide" label, not just the href —
+    // a blanked label would still leave the anchor above. The label renders in
+    // its own <span> (after the icon), so assert it there rather than as loose
+    // page text that unrelated copy could satisfy.
+    expect(html).toContain("<span>Questions guide</span>");
+  });
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/ui/templates/admin/questions/list.test.ts]: Fixed in 99edbd1 — split into "marks the questions nav entry active", "renders the add-question form", and "links to the questions guide", the last keeping the `<span>Questions guide</span>` assertion and its comment.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/ui/templates/checkin/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  quantity = 1,
+  phone = "",
+): Promise<{ listing: Listing; session: CheckinSession; token: string }> => {
+  const { listing, token } = await createTestAttendeeWithToken(
+    name,
+    email,
+    listingOverrides,
+    quantity,
+    phone,
+  );
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/ui/templates/checkin/helpers.ts:28]: Fixed in 99edbd1 — added `type ListingOverrides = Parameters<typeof createTestAttendeeWithToken>[2]` and typed the parameter with it, so excess properties are caught at the call site instead of being widened to `{}`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/ui/templates/checkin/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import { mockFormRequest } from "#test-utils/mocks.ts";
+import { testCookie, testCsrfToken } from "#test-utils/session.ts";
+
+interface CheckinSession {
+  cookie: string;
+  csrfToken: string;
+}
+
+/** Create attendee + login, returning token + session for check-in tests */
+type ListingOverrides = Parameters<typeof createTestAttendeeWithToken>[2];
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/ui/templates/checkin/helpers.ts:13]: Correct, and my mistake when applying the earlier type suggestion — the alias landed between the JSDoc and the function it documents. Fixed locally by moving the alias above the comment, so the comment sits on `setupCheckinTest` again.

The push is currently rejected because the PR is in the merge queue and the branch is locked. If it merges as-is I'll open a small follow-up PR with just this move; if it leaves the queue I'll push it here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/features/admin/auth/sessions.test.ts [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+      await createSession("other", "csrf-other", Date.now() + 10000, null, 1);
+
+      const { cookie, csrfToken } = await loginAsAdmin();
+
+      // Extract the session token from cookie
+      const sessionMatch = cookie.match(
+        new RegExp(`${getSessionCookieName()}=([^;]+)`),
+      );
+      // loginAsAdmin always issues a session cookie, so the match is guaranteed.
+      const sessionToken = sessionMatch![1]!;
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/features/admin/auth/sessions.test.ts:140]: Not taking this one — it asks for the opposite of the repo's stated rule, and of the change that was just made here at CodeRabbit's request on this same line.

AGENTS.md "Trust application invariants" and "Don't defend against the impossible" say not to guard states a foundational system already guarantees, and the recorded learning from PR #1681 is explicit: when a value is genuinely guaranteed by application logic, prefer a non-null assertion with a comment documenting the invariant over a defensive throw, because the throw is untestable dead code and a coverage gap.

`loginAsAdmin()` issuing a session cookie is exactly that kind of invariant — it is the helper's entire job, and if it stopped doing it, dozens of tests in this file and elsewhere would fail first and far more loudly than a bespoke throw here. The comment above the assertion records why it holds.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: test/ui/templates/admin/questions/answer.test.ts [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+import {
+  adminAnswerEditPage,
+  adminAnswerRecalculatePage,
+} from "#templates/admin/questions.tsx";
+import { resetFeaturePageTest } from "#test/ui/templates/admin/feature-page-test.ts";
+import { OWNER_SESSION } from "#test-utils/admin-page-test.ts";
+import { testAnswer, testQuestion } from "#test-utils/factories.ts";
+import { setupQuestionPageTest, tShirtQuestion } from "./fixtures.ts";
+
+describe("adminAnswerEditPage", () => {
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply test/ui/templates/admin/questions/answer.test.ts:12]: The mechanism is described correctly — `ownsTest` in `scripts/mutation/test-map.ts` matches `prefix/`, so a `questions.tsx` mutant in the automatic gate selects all six files and `execution.ts` runs them with `--parallel`, i.e. more isolates than the single file it replaced. Two reasons I'm not changing the mapping here.

First, the guideline this PR is delivering is about the manual command, which is the one you point at a module while hardening it: `deno task mutation src/ui/templates/admin/questions.tsx 'test/ui/templates/admin/questions/answer.test.ts'` was impossible before this split and is now the narrow run the AGENTS.md line describes. That capability is what the split buys.

Second, changing the automatic gate's selection or generating a combined entry file is a change to the mutation tooling's design, not to the tests being moved — exactly the kind of scope creep that would bury a mechanical test move in unrelated machinery. It also needs its own measurement to justify: whether N small isolates actually cost more than one large one depends on the per-isolate module-graph cost against the smaller per-file suites, and I have not measured that.

I've recorded it in `TODO.md` under the existing mutation follow-ups so the mapping question is not lost.

---
_Generated by [Claude Code](https://claude.ai/code)_
